modules/electromagnetics/include/bcs/VectorEMRobinBC.h

```diff
@@ -0,0 +1,60 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "VectorIntegratedBC.h"
+#include <complex>
+
+/**
+ *  First order Robin-style Absorbing/Port boundary condition for vector nonlinear
+ *  variables.
+ */
+class VectorEMRobinBC : public VectorIntegratedBC
+{
+public:
+  static InputParameters validParams();
+
+  VectorEMRobinBC(const InputParameters & parameters);
+
+protected:
+  virtual Real computeQpResidual() override;
+  virtual Real computeQpJacobian() override;
+  virtual Real computeQpOffDiagJacobian(unsigned int jvar) override;
+
+  /// Scalar waveguide propagation constant
+  const Function & _beta;
+
+  /// Variable field component (real or imaginary)
+  const MooseEnum _component;
+
+  /// Coupled field vector variable
+  const VectorVariableValue & _coupled_val;
+
+  /// Coupled field vector variable id
+  unsigned int _coupled_var_num;
+
+  /// Real incoming field vector
+  const Function & _inc_real;
+
+  /// Imaginary incoming field vector
+  const Function & _inc_imag;
+
+  /// Value of complex number (can't use _j or _i due to basis function indices)
+  const std::complex<double> _jay;
+
```

Comment:
should we define this at a real high level like a M_PI

Reply:
@lindsayad made a good suggestion about putting this in the electromagnetics module namespace that I already use for the BC and component enums. The complex number "j" needs to exist at the module level, but I don't think we need to go up as high as framework. As far as I understand, my module is the only one that needs that at the moment.